scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -1767,22 +1773,31 @@ class FormatOps(
   }
 
   // Redundant () delims around case statements
-  def isCaseBodyEnclosedAsBlock(ft: FormatToken, caseStat: Case)(implicit
+  def isCaseBodyEnclosedAsBlock(ft: FormatToken, caseStat: Tree)(implicit
       style: ScalafmtConfig
   ): Boolean = {
-    val body = caseStat.body
-    (ft.noBreak || style.newlines.getBeforeMultiline.ignoreSourceSplit) &&
-    body.eq(ft.meta.rightOwner) && (body match {
-      case _: Lit.Unit | _: Term.Tuple => false
-      case t: Term.ApplyInfix =>
-        val op = t.op.value
-        op != "->" && op != "→"
-      case _ => true
-    }) && {
-      val btoks = body.tokens
-      btoks.headOption.exists { head =>
-        head.is[Token.LeftParen] && matchingOpt(head).contains(btoks.last)
+    val bodyOpt = caseStat match {
+      case tc: TypeCase => Some(tc.body)
+      case c: Case => Some(c.body)
+      case _ => None
+    }
+    if (bodyOpt.isDefined) {
```

Comment:
may i suggest replacing with a functional
```
bodyOpt.exists { body =>
  ...
}
```

another option is to preserve the original code virtually unchanged, use the fact that `body.eq(rightOwner)` is part of the logical expression and simply use `null` for body (or anything else that's not `rightOwner`) for non-case trees.

Reply:
Fixed! Added exists.